### PR TITLE
Add NV12 support

### PR DESCRIPTION
**Pull request by mwang30**
Jira: None.
Test: can load and show nv12 format picture

Signed-off-by: Mingwei Wang <mingwei.wang@intel.com>

### CODE REVIEW

File: i915.c [hunk 1]
```diff
@@ -468,7 +468,7 @@ static uint32_t i915_resolve_format(uint32_t format)
 		/*HACK: See b/28671744 */
 		return DRM_FORMAT_XBGR8888;
 	case DRM_FORMAT_FLEX_YCbCr_420_888:
-		return DRM_FORMAT_YVU420;
+		return DRM_FORMAT_NV12;
```
Comment (kalyankondapally) [inline i915.c]: hmm with this sw decoded video will be broken right ?

Comment (mwang30) [inline i915.c]: That's Tapani's suggestion here:
https://github.com/android-ia/external-mesa/commit/731c5e3a2eecff72ff85f7e4de0a17a6eec4133d#commitcomment-21573614
We need more discussion to align.

Comment (kalyankondapally) [inline i915.c]: seems fine actually, can you just check sw decoded video is not broken after this

**Comment by mwang30** [comment]
Guangxin tried video playback with this patch, it works.

**Comment by xuguangxin** [comment]
Hi @kalyankondapally 
https://github.com/android-ia/device-androidia/pull/65 depends on this. And we are planned to enable video playback at WW19.5.  
Any update on this?

File: i915.c [hunk 2]
```diff
@@ -453,7 +453,7 @@ static uint32_t i915_resolve_format(uint32_t format)
 		/*HACK: See b/28671744 */
 		return DRM_FORMAT_XBGR8888;
 	case DRM_FORMAT_FLEX_YCbCr_420_888:
-		return DRM_FORMAT_YVU420_ANDROID;
+		return DRM_FORMAT_NV12;
```
Comment (kalyankondapally) [inline i915.c]: Instead of this, I think we should return this for PLATFORM_dEFINED and usage being for Video.

**Comment by xuguangxin** [comment]
Hi @mwang30 , any update on this?

**Comment by kalyankondapally** [comment]
@mwang30 Any update on this ?

**Comment by mwang30** [comment]
@kalyankondapally I didn't catch your point. I saw rockchip.c return DRM_FORMAT_NV12 too.

**Comment by kalyankondapally** [comment]
I see this error:
"06-04 22:19:53.981 26399 26518 E GLConsumer: error creating EGLImage: 0x3000
06-04 22:19:53.981 26399 26518 E GLConsumer: Failed to create image. size=640x360 st=640 usage=0x2130 fmt=842094169
06-04 22:19:53.981 26399 26518 W GLConsumer: [SurfaceTexture-179-26399-1] updateAndRelease: unable to createImage on display=0x1 slot=0
06-04 22:19:53.982 26399 26518 E SurfaceTexturePlatformWrapper: Error calling updateTexImage
06-04 22:19:53.982 26399 26518 E SurfaceTexturePlatformWrapper: java.lang.RuntimeException: Error during updateTexImage (see logcat for details)
06-04 22:19:53.982 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.nativeUpdateTexImage(Native Method)
06-04 22:19:53.982 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.updateTexImage(SurfaceTexture.java:244)
06-04 22:19:53.982 26399 26518 E SurfaceTexturePlatformWrapper: 	at org.chromium.ui.gl.SurfaceTexturePlatformWrapper.updateTexImage(SurfaceTexturePlatformWrapper.java:43)
06-04 22:19:54.014 26399 26518 E GLConsumer: error creating EGLImage: 0x3000
06-04 22:19:54.014 26399 26518 E GLConsumer: Failed to create image. size=640x360 st=640 usage=0x2130 fmt=842094169
06-04 22:19:54.014 26399 26518 W GLConsumer: [SurfaceTexture-179-26399-1] updateAndRelease: unable to createImage on display=0x1 slot=0
06-04 22:19:54.014 26399 26518 E SurfaceTexturePlatformWrapper: Error calling updateTexImage
06-04 22:19:54.014 26399 26518 E SurfaceTexturePlatformWrapper: java.lang.RuntimeException: Error during updateTexImage (see logcat for details)
06-04 22:19:54.014 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.nativeUpdateTexImage(Native Method)
06-04 22:19:54.014 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.updateTexImage(SurfaceTexture.java:244)
06-04 22:19:54.014 26399 26518 E SurfaceTexturePlatformWrapper: 	at org.chromium.ui.gl.SurfaceTexturePlatformWrapper.updateTexImage(SurfaceTexturePlatformWrapper.java:43)
06-04 22:19:54.047 26399 26518 E GLConsumer: error creating EGLImage: 0x3000
06-04 22:19:54.047 26399 26518 E GLConsumer: Failed to create image. size=640x360 st=640 usage=0x2130 fmt=842094169
06-04 22:19:54.047 26399 26518 W GLConsumer: [SurfaceTexture-179-26399-1] updateAndRelease: unable to createImage on display=0x1 slot=0
06-04 22:19:54.048 26399 26518 E SurfaceTexturePlatformWrapper: Error calling updateTexImage
06-04 22:19:54.048 26399 26518 E SurfaceTexturePlatformWrapper: java.lang.RuntimeException: Error during updateTexImage (see logcat for details)
06-04 22:19:54.048 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.nativeUpdateTexImage(Native Method)
06-04 22:19:54.048 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.updateTexImage(SurfaceTexture.java:244)
06-04 22:19:54.048 26399 26518 E SurfaceTexturePlatformWrapper: 	at org.chromium.ui.gl.SurfaceTexturePlatformWrapper.updateTexImage(SurfaceTexturePlatformWrapper.java:43)
06-04 22:19:54.081 26399 26518 E GLConsumer: error creating EGLImage: 0x3000
06-04 22:19:54.081 26399 26518 E GLConsumer: Failed to create image. size=640x360 st=640 usage=0x2130 fmt=842094169
06-04 22:19:54.081 26399 26518 W GLConsumer: [SurfaceTexture-179-26399-1] updateAndRelease: unable to createImage on display=0x1 slot=0
06-04 22:19:54.081 26399 26518 E SurfaceTexturePlatformWrapper: Error calling updateTexImage
06-04 22:19:54.081 26399 26518 E SurfaceTexturePlatformWrapper: java.lang.RuntimeException: Error during updateTexImage (see logcat for details)
06-04 22:19:54.081 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.nativeUpdateTexImage(Native Method)
06-04 22:19:54.081 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.updateTexImage(SurfaceTexture.java:244)
06-04 22:19:54.081 26399 26518 E SurfaceTexturePlatformWrapper: 	at org.chromium.ui.gl.SurfaceTexturePlatformWrapper.updateTexImage(SurfaceTexturePlatformWrapper.java:43)
06-04 22:19:54.114 26399 26518 E GLConsumer: error creating EGLImage: 0x3000
06-04 22:19:54.114 26399 26518 E GLConsumer: Failed to create image. size=640x360 st=640 usage=0x2130 fmt=842094169
06-04 22:19:54.114 26399 26518 W GLConsumer: [SurfaceTexture-179-26399-1] updateAndRelease: unable to createImage on display=0x1 slot=0
06-04 22:19:54.115 26399 26518 E SurfaceTexturePlatformWrapper: Error calling updateTexImage
06-04 22:19:54.115 26399 26518 E SurfaceTexturePlatformWrapper: java.lang.RuntimeException: Error during updateTexImage (see logcat for details)
06-04 22:19:54.115 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.nativeUpdateTexImage(Native Method)
06-04 22:19:54.115 26399 26518 E SurfaceTexturePlatformWrapper: 	at android.graphics.SurfaceTexture.updateTexImage(SurfaceTexture.java:244)
06-04 22:19:54.115 26399 26518 E SurfaceTexturePlatformWrapper: 	at org.chromium.ui.gl.SurfaceTexturePlatformWrapper.updateTexImage(SurfaceTexturePlatformWrapper.java:43)
"

**Comment by raghs7** [comment]
Can you please rebase ?

**Comment by raghs7** [comment]
disregard the comment to rebase, i see this is rebased as #11

**Pull request closed without merging**